platform-core-module/src/main/java/com/softicar/platform/core/module/file/smb/jcifs/JcifsSmbDirectory.java

```diff
@@ -37,6 +41,28 @@ public JcifsSmbDirectory(String context, String name, NtlmPasswordAuthentication
 		}
 	}
 
+	@Override
+	public Collection<String> listFilesRecursively(String root) {
```

Comment:
Something went wrong here - `listFilesRecursively` and `listFiles` should not need to be so similar, and complicated.

This might be sufficient to walk the directory tree, in depth-first order. Did not test it though.
```
@Override
public Collection<String> listFilesRecursively(String subDirectory) {

	List<String> fileNames = new ArrayList<>();
	for (ISmbFile file: getSubDirectory(subDirectory).listFiles()) {
		if (file.isDirectory()) {
			fileNames.addAll(listFilesRecursively(subDirectory + "/" + file.getName()));
		} else {
			fileNames.add(file.getCanonicalPath());
		}
	}
	return fileNames;
}
```

Reply:
I dont think this will work, maybe im just confused because its recursion, but with the way you suggest, I have a feeling it wont work when I look at `StoredFileSmbContentStore` 
Hard to explain via text, but I reverted some other changes from me so that the implementation is now nearly the same as it was in `StoredFileSmbContentStore` which should reduce the chance that something broke

Reply:
Doesnt work, fixed and tested my version